lib/presentation/view/popular_movies_list_view.dart

```diff
@@ -0,0 +1,133 @@
+import 'package:flutter/material.dart';
+
+import '../../core/util/constants/ui_constants.dart';
+import '../../core/util/text_styles/custom_text_style.dart';
+import '../../data/model/data_state.dart';
+import '../bloc/now_playing_movies_bloc.dart';
+import '../bloc/popular_movies_bloc.dart';
+import '../bloc/top_rated_movies_bloc.dart';
+import '../bloc/upcoming_movies_bloc.dart';
+import '../widget/drawer_tiles.dart';
+import '../widget/movie_info.dart';
+import 'now_playing_movies_view.dart';
+import 'top_rated_movies_view.dart';
+import 'upcoming_movies_view.dart';
+
+class PopularMoviesList extends StatefulWidget {
+  const PopularMoviesList({super.key, required this.bloc});
+
+  final PopularMoviesBloc bloc;
+  @override
+  State<PopularMoviesList> createState() => _PopularMoviesListState();
+}
+
+class _PopularMoviesListState extends State<PopularMoviesList> {
+  static const int crossAxisCount = 2;
+  static const double childAspectRatio = 0.7;
+  static const double mainAxisSpacing = 5;
+  static const String errorMessage = 'An error has occurred while loading data';
+  static const String scaffoldKey = 'scaffold2';
+
+  @override
+  void initState() {
+    widget.bloc.getPopularMovies();
+    super.initState();
+  }
+
+  @override
+  void didUpdateWidget(covariant PopularMoviesList oldWidget) {
+    widget.bloc.getPopularMovies();
+    super.didUpdateWidget(oldWidget);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      key: const Key(scaffoldKey),
+      backgroundColor: Colors.black12,
+      appBar: AppBar(
+        title: const Center(
+          child: Text(
+            MovieDetailsUiConstants.popularMoviesLabel,
+          ),
+        ),
+      ),
+      body: StreamBuilder<DataState>(
+        initialData: widget.bloc.initialData,
+        stream: widget.bloc.popularMovies,
+        builder: (context, snapshot) {
```

Comment:
Add the type of the parameter that you are using `(BuildContext and AsyncSnapshot<DataState>)`

Reply:
done